core/src/main/boilerplate/tapir/typelevel/FnComponents.scala.template

```diff
@@ -0,0 +1,29 @@
+package tapir.typelevel;
+
+/*
+ * FN.tupled = TUPLE => RESULT
+ */
+trait FnComponents[-FN, TUPLE, RESULT] {
+  def tupled(f: FN): TUPLE => RESULT
+  def arity: Int
+}
+
+object FnComponents {
+
+  private implicit class HackAddTupledMethodToFunction1[-T1, +R](f: T1 => R) {
+    def tupled: Tuple1[T1] => R = {
+      case Tuple1(x1) => f.apply(x1)
+    }
+  }
+
+  implicit def fn1_[T1, R]: FnComponents[T1 => R, T1, R] = new FnComponents[T1 => R, T1, R] {
```

Comment:
won't there be two instances for `T1=>R` (function with 1 argument)?

Reply:
No, because the second one takes Tuple1 as a first argument

Reply:
Hmm, you might be right :)